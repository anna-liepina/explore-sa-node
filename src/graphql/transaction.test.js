describe('GraphQL: Transaction', () => {
    describe('transaction', () => {
        it(`should fetch transaction by ID`, async () => {
            const { data: { transaction: data } } = await query({
                query: `
                {
                    transaction(id: 1) {
                        id
                        price
                        date
                    }
                }`
            });

            expect(data).toMatchSnapshot();
        });
    });

    describe('transactionSearch [presets: perPage 3]', () => {
        it(`search without date range`, async () => {
            const { data: { transactionSearch: data } } = await query({
                query: `
                {
                    transactionSearch(perPage: 3) {
                        id
                        price
                        date
                    }
                }`
            });

            expect(data).toMatchSnapshot();
        });

        it(`search in data range, first page, 3 per page`, async () => {
            const { data: { transactionSearch: data } } = await query({
                query: `
                {
                    transactionSearch(from: "2010-01-01", to: "2021-01-01", perPage: 3) {
                        id
                        price
                        date
                    }
                }`
            });

            expect(data).toMatchSnapshot();
        });

        it(`search in data range, seconds page, 3 per page`, async () => {
            const { data: { transactionSearch: data } } = await query({
                query: `
                {
                    transactionSearch(from: "2010-01-01", to: "2021-01-01", page: 2, perPage: 3) {
                        id
                        price
                        date
                    }
                }`
            });

            expect(data).toMatchSnapshot();
        });
    });
});
