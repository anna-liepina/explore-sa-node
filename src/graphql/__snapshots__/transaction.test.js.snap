// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphQL: Transaction transaction should fetch transaction by ID 1`] = `
Object {
  "date": "2019-04-26",
  "id": "1",
  "price": 375000,
}
`;

exports[`GraphQL: Transaction transactionSearch [presets: perPage 3] within date range 1`] = `
Array [
  Object {
    "date": "2013-10-23",
    "id": "65",
    "price": 100000,
  },
  Object {
    "date": "2013-12-03",
    "id": "67",
    "price": 130000,
  },
  Object {
    "date": "2013-12-18",
    "id": "18",
    "price": 125000,
  },
]
`;

exports[`GraphQL: Transaction transactionSearch [presets: perPage 3] within date range: 2nd page 1`] = `
Array [
  Object {
    "date": "2015-10-15",
    "id": "455",
    "price": 765000,
  },
  Object {
    "date": "2015-10-30",
    "id": "431",
    "price": 757055,
  },
  Object {
    "date": "2015-10-30",
    "id": "453",
    "price": 745000,
  },
]
`;

exports[`GraphQL: Transaction transactionSearch [presets: perPage 3] without criteria 1`] = `
Array [
  Object {
    "date": "2007-08-22",
    "id": "36",
    "price": 229400,
  },
  Object {
    "date": "2013-10-23",
    "id": "65",
    "price": 100000,
  },
  Object {
    "date": "2013-12-03",
    "id": "67",
    "price": 130000,
  },
]
`;
